src/app/shared/entities/location/location-reducer.ts

```diff
@@ -15,24 +15,32 @@ export const reducer = createReducer(
     on(
         LocationActions.selectLocation,
         (state, {location}) => {
-            return locationEntityAdapter.updateOne({
+            const updatedState = locationEntityAdapter.updateOne({
                 id: location._id,
                 changes: {
                     selected: true
                 }
             }, state);
+            const map = localStorage.getItem('map')
+            localStorage.setItem(`${map}-locations`, JSON.stringify(updatedState))
```

Comment:
This will be more a general comment. I've been running on the line of thought that Reducers should mostly act as pure functions and this, I feel, causes a side-effect that could be unintended when just interacting with the store. IMO Reducers really should be used to update/calculate the new state and pass that on. Interacting with the local storage should be handled as a side-effect somewhere else (See my original big-wall-o-text on how I was approaching this).

Although this would solve the save/hydrate with local storage it doesn't necessarily address the bigger concern/solution about avoiding making extra API calls when it doesn't need to. Which that logic should be handled in the effect, or that's how I've been doing it in my experience with NGRX. Admittedly I totally left that out and the effect was always making API calls it didn't need to and mostly neglecting the store as a real 'caching' solution.